### ISSUE TITLE
openstack-cloud-controller-manager-binary does not show the correct release in OKD/OCP4.12+

### ISSUE THREAD

**Issue by jonasbartho** [kubernetes/cloud-provider-openstack#2623]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
The `openstack-cloud-controller-manager` binary in OKD 4.12(=free version of Openshift) does not show the correct git-release that is being used. 

`# openstack-cloud-controller-manager --version`
`Kubernetes v0.0.0-master+$Format:%H$`

**What you expected to happen**:
Since `openstack-cloud-controller-manager` is the out-of-tree provider of Openstack by default in OKD/OCP4.12+, it would be nice to see which version it is using. I for example want to use new functionality from `release-1.30`, but it looks like it is not available yet. Is there any other way to check which release I am on?

**Environment**:
- openstack-cloud-controller-manager version: unknown..
- OpenStack version: 2024.1
- Others: OKD/OCP 4.15

**Comment by jonasbartho** [comment]
I found the following command that gives sort-of a hunch which version currently is being used:

```
$ oc adm release info --commit-urls |grep -i openstack-cloud
  openstack-cloud-controller-manager             https://github.com/openshift/cloud-provider-openstack/commit/e1a9f48173b0056dc96b62ef6b141ccc1838cebc
```

**Comment by jichenjc** [comment]
>>># openstack-cloud-controller-manager --version
Kubernetes v0.0.0-master+$Format:%H$

I remember it's a known issue and somehow related to Cloud provider instead of OCCM
but not 100% sure now

**Comment by kayrus** [comment]
I remember that I fixed this last year in #2252
